tests/integration.test.ts

```diff
@@ -1,126 +1,258 @@
+  handleUpdatePodAdminV1_4,
+  handleDeregisterPodV1_3,
+  handleDeregisterPodV1_4,
+  handleTransferBatch,
+} from "../src/mapping";
+import {
+  generateDeregisterPodV1_3,
+  generateDeregisterPodV1_4,
+  generateTransferSingle,
+  generateTransferBatch,
+  generateUpdatePodAdmin,
+  generateUpdatePodAdminV1,
+} from "./eventGenerators";
 import { log } from "matchstick-as/assembly/log";
 import { User, PodUser } from "../generated/schema";
 import { addressZero, addressOne, addressTwo, addressThree } from "./fixtures";
 
-test('TransferSingle should create a Pod, User, and UserPod entity', () => {
-  // Mint a token (transfer from zero to addressTwo)
-  let transferSingleEvent = generateTransferSingle(
-    addressOne,
-    addressZero,
-    addressOne,
-    1,
-    1,
-  );
-
-  handleTransferSingle(transferSingleEvent);
-  assert.fieldEquals('Pod', '1', 'id', '1');
-  // Admin should always be null for a fresh pod on creation.
-  // Updating should be handled by UpdatePodAdmin, not this function.
-  assert.fieldEquals('Pod', '1', 'admin', addressZero)
-  assert.fieldEquals('User', addressOne, 'id', addressOne);
-  assert.fieldEquals('PodUser', addressOne + '-1', 'user', addressOne);
-  assert.fieldEquals('PodUser', addressOne + '-1', 'pod', '1');
-  clearStore();
-});
+describe("TransferSingle", () => {
+  test("TransferSingle should create a Pod, User, and UserPod entity", () => {
+    // Mint a token (transfer from zero to addressTwo)
+    let transferSingleEvent = generateTransferSingle(
+      addressOne,
+      addressZero,
+      addressOne,
+      1,
+      1
+    );
 
-test('TransferSingle should remove the existing UserPod entity when token is transferred out', () => {
-  let fromUser = new User(addressOne);
-  fromUser.save();
-  let userPod = new PodUser(addressOne + '-1');
-  userPod.save();
-
-  // Transfer token from AddressOne to addressTwo
-  let transferSingle = generateTransferSingle(
-    addressOne,
-    addressOne,
-    addressTwo,
-    1,
-    1,
-  );
-  
-  handleTransferSingle(transferSingle);
-  assert.fieldEquals('PodUser', addressTwo + '-1', 'user', addressTwo);
-  assert.fieldEquals('PodUser', addressTwo + '-1', 'pod', '1');
-  assert.notInStore('PodUser', addressOne + '-1');
-  clearStore();
-});
+    handleTransferSingle(transferSingleEvent);
+    assert.fieldEquals("Pod", "1", "id", "1");
+    // Admin should always be null for a fresh pod on creation.
+    // Updating should be handled by UpdatePodAdmin, not this function.
+    assert.fieldEquals("Pod", "1", "admin", addressZero);
+    assert.fieldEquals("User", addressOne, "id", addressOne);
+    assert.fieldEquals("PodUser", addressOne + "-1", "user", addressOne);
+    assert.fieldEquals("PodUser", addressOne + "-1", "pod", "1");
+    clearStore();
+  });
+
+  test("TransferSingle should remove the existing UserPod entity when token is transferred out", () => {
+    let fromUser = new User(addressOne);
+    fromUser.adminPods = [addressTwo];
```

Comment:
🎨 I did have to add lines like this though, because the existing tests were failing without the recently required attributes.